.circleci/config.yml

```diff
@@ -0,0 +1,111 @@
+version: 2.1
+
+orbs:
+  codacy: codacy/base@1.2.1
+
+references:
+  default_golang_image: &default_golang_image
+    docker:
+      - image: golang:1.13.7-alpine3.11
+    working_directory: ~/workdir/
+
+  get_requirements: &get_requirements
+    run:
+      name: Install requirements
+      command: |
+        apk add git alpine-sdk
+        go get github.com/pborman/getopt
+        go get github.com/stretchr/testify/assert
+  attach_to_workspace: &attach_to_workspace
+    attach_workspace:
+      at: ~/
+
+  persist_to_workspace: &persist_to_workspace
+    persist_to_workspace:
+      root: ~/
+      paths:
+        - workdir
+jobs:
+  checkout_and_version:
+    docker:
+      - image: codacy/git-version:1.0.7
+    working_directory: ~/workdir
+    steps:
+      - checkout
+      - run:
+          name: Set version
+          command: /bin/git-version > VERSION
+      - run:
+          name: Current version
+          command: cat VERSION
+      - <<: *persist_to_workspace
+
+  test:
+    <<: *default_golang_image
+    steps:
+      - <<: *attach_to_workspace
+      - <<: *get_requirements
+      - run:
+          name: "Run tests"
+          command: go test ./...
+      - <<: *persist_to_workspace
+
+  build:
+    <<: *default_golang_image
+    steps:
+      - <<: *attach_to_workspace
+      - <<: *get_requirements
+      - run:
+          name: "Build"
+          command: go build .
+      - <<: *persist_to_workspace
+
+  tag_version:
+    docker:
+      - image: codacy/git-version:1.0.7
+    working_directory: ~/workdir
+    steps:
+      - <<: *attach_to_workspace
+      - add_ssh_keys:
+          fingerprints:
+            - "df:83:d7:c7:d5:79:06:c2:3b:d1:fd:e2:a3:d1:12:c5"
+      - run:
+          name: add github known host
+          command: |
+            mkdir -p ~/.ssh
+            touch ~/.ssh/known_hosts
+            ssh-keyscan github.com >> ~/.ssh/known_hosts
+      - deploy:
+          name: Edit version in plugin.yaml
+          command: |
+            sed -E "s/version: \"[0-9].*\"/version: \"$(cat VERSION)\"/g" plugin.yaml.new
+            rm plugin.yaml
+            mv plugin.yaml.new plugin.yaml
+      - deploy:
+          name: Tag git repository
+          command: git tag $(cat VERSION) && git push --tags
+
+workflows:
+  test:
+    jobs:
+      - checkout_and_version
+      - test:
+          requires:
+            - checkout_and_version
+
+  deploy:
+    jobs:
+      - checkout_and_version:
+          filters:
+            branches:
+              only:
+                - master
+      - test:
+          requires:
+            - checkout_and_version
```

Comment:
Since these two steps are already defined for test workflow you can have a single workflow where the filter is only in the `build` job